src/components/authorize/authorize-controller.ts

```diff
@@ -166,7 +151,13 @@ export function authorizeGet(
       const consentCookieValue =
         cookieService.createConsentCookieValue(cookieConsent);
 
-      createConsentCookie(res, consentCookieValue);
+      res.cookie(COOKIES_PREFERENCES_SET, consentCookieValue.value, {
+        expires: consentCookieValue.expiry,
+        secure: true,
+        httpOnly: false,
+        domain: res.locals.analyticsCookieDomain,
+        encode: String,
+      });
```

Comment:
Hi @andrew-moores - I've introduced an additional commit that extends the tests to assert that calls to `res.cookie` include the expected arguments. I initially had a bit of difficulty because the `expiry` property was changing between calls (by a second) and causing failures, but found I could assert for only the `httpOnly`, `secure` and `encode` properties by passing these as keys to an object passed to `sinon.match`

Reply:
I've added some suggestions for testing the expiry date as it seems like an important feature to check.

Reply:
Thanks @andrew-moores - I've included all the suggestions now.